packages/buckets/src/buckets.ts

```diff
@@ -247,10 +283,42 @@ export class Buckets extends BucketsGrpcClient {
    * Returns information about a bucket path.
    * @param key Unique (IPNS compatible) identifier key for a bucket.
    * @param path A file/object (sub)-path within a bucket.
+   * @param depth (optional) will walk the entire bucket to target depth (default = 1)
    */
-  async listPath(key: string, path: string): Promise<ListPathReply.AsObject> {
+  async listPath(key: string, path: string, depth = 1): Promise<ListPathReply.AsObject> {
```

Comment:
switched it to depth, so that it's not going to walk the full directory all the time. instead you need to manually tell it how deep to go. feels a bit safer than unbounded recursive, thoughts?

Reply:
Is setting depth to 0 going to be essential infinite? This seems reasonable to me.

Reply:
not really. depth 0 will return the same as depth 1 because the api always returns leafs within 1 order down. -1 is unbounded.